### PR TITLE
Single product page

### PR DESCRIPTION
**Pull request by bajajcodes**
## Single Product Page
- Get product info with one click, for desktop view horizontal card is used and for mobile view vertical card.
- On page refresh data is not persisted. So, might get product does not exists in database, meanwhile it exists.

## Note
- Did not used git in correct way, hence files changed are large in number.
- Ignore others files and review files mentioned below.

## What Files to check
- RouteSwitch.js
- components/imageslider.js
- context/products/filters/
- pages/products/aside/
- pages/products/useGetQueryProducts.js
- pages/singleproduct/

**Total files to check: 7**

## Deploy Preview
https://deploy-preview-15--3on3store-react.netlify.app/

### CODE REVIEW

**Review by RakshithBhat03 — APPROVED** [review]
- The code is clean and well structured.
- The components are abstracted well.

File: src/context/products/filters/filters.helpers.js [hunk 1]
```diff
@@ -94,5 +94,17 @@ async function getProducts() {
...
 
+async function getProduct(productId) {
+  try {
+    const response = await axios.get(`/api/products/${productId}`);
+    if (response.status !== 200) {
+      throw new Error("Getting product failed, Invalid product credentials");
+    }
+    const product = response.data?.product;
+    return { product, exception: null };
+  } catch (exception) {
```
Comment (koushikrp) [inline src/context/products/filters/filters.helpers.js:105]: Good Error Handling.

File: src/pages/auth/main/useSignupValidation.js [hunk 2]
```diff
@@ -5,9 +5,9 @@ import {
   togglePasswordInputType,
   getUserSignup,
 } from "./main.helper";
-import { useAuthContext } from "context";
+import { useAuthContext, useAlert } from "context";
 
-export function useSignupValidation(setAlertInfo) {
+export function useSignupValidation() {
```
Comment (koushikrp) [inline src/pages/auth/main/useSignupValidation.js:10]: Good Naming Convention.

**Review by koushikrp — APPROVED** [review]
Best Coding Practices are followed. Good to merge.

**Pull request merged**